src/main/java/com/odysseusinc/arachne/portal/service/analysis/impl/ExecutableDetector.java

```diff
@@ -0,0 +1,14 @@
+package com.odysseusinc.arachne.portal.service.analysis.impl;
+
+import com.odysseusinc.arachne.commons.api.v1.dto.CommonAnalysisType;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+@Component
+public class ExecutableDetector {
+
+    public boolean detectExecutable(CommonAnalysisType type, MultipartFile file) {
```

Comment:
I think that `isFileExecutable` is better fit here

Reply:
can be